File: tests/test_code/py/ambiguous_resolution/ambiguous_resolution.py
class Abra():
    def magic():
        pass

    def abra_it():
        pass


class Cadabra():
    def magic():
        pass

    def cadabra_it(a=None):
        a.abra_it()


def main(cls):
    obj = cls()
    obj.magic()
    obj.cadabra_it()


main()

File: tests/test_code/rb/ambiguous_resolution/ambiguous_resolution.rb
class Abra
    def magic()
    end

    def abra_it()
    end
end

class Cadabra
    def magic()
    end

    def cadabra_it(a: nil)
        a.abra_it()
    end
end

def main(cls)
    obj = cls.new()
    obj.magic()
    obj.cadabra_it()
end

main(Cadabra)
